src/QuadForm/Herm/Lattices.jl

```diff
@@ -16,136 +16,103 @@ end
 #
 ################################################################################
 
+function lattice(V::HermSpace, B::PMat)
+  E = base_ring(V)
+  @req rank(matrix(B)) == min(nrows(B), ncols(B)) "B must be of full rank"
```

Comment:
This seems quite expensive. I think we should allow to turn this check of by using
`lattice(::HermSpace, B::PMat, check::Bool=True)`
Any other suggestion @thofma ?

Reply:
Eh, we can disable it with a check yes. Makes sense.